### PR TITLE
make sure position_ids are passed in for causal mask creation for gpt-oss

### PR DESCRIPTION
**Pull request by winglian**
# What does this PR do?

Packing won't work with gpt-oss since it doesn't respect the position ids. See https://github.com/huggingface/transformers/pull/39194

Fixes # (issue)

@ArthurZucker @Cyrilvallez 

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#create-a-pull-request),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

File: src/transformers/models/gpt_oss/modeling_gpt_oss.py [hunk 1]
```diff
@@ -478,6 +478,7 @@ def forward(
                 "attention_mask": attention_mask,
                 "cache_position": cache_position,
                 "past_key_values": past_key_values,
+                "position_ids": position_ids,
```
Comment (ArthurZucker) [inline src/transformers/models/gpt_oss/modeling_gpt_oss.py:481]: ty can you update the `modular` as well ?!

**Comment by Cyrilvallez** [comment]
Indeed, needs to be upstreamed in modular as well!
